base/util/src/main/java/com/netscape/cmsutil/crypto/CryptoUtil.java

```diff
@@ -805,25 +805,48 @@ public static KeyPair generateECCKeyPair(
     }
 
     public static void setClientCiphers(String list) throws SocketException {
+          setClientCiphers(null, list);
+    }
+    public static void setClientCiphers(SSLSocket soc, String list) throws SocketException {
+        String method = "CryptoUtil.setClientCiphers:";
+        if (soc == null)
+            logger.debug(method + "begins");
+        else
+            logger.debug(method + "on soc begins");
 
         if (list == null) {
+            logger.debug(method + "no cipher list in call; using default");
             // use default
-            setDefaultSSLCiphers();
+            if (soc == null)
+                setDefaultSSLCiphers();
             return;
         }
 
+        logger.debug(method + "cipher list in call; processing...");
         String ciphers[] = list.split(",");
         if (ciphers.length == 0) return;
 
-        unsetSSLCiphers();
+        if (soc == null)
+            unsetSSLCiphers();
+        else
+            unsetSSLCiphers(soc);
 
         for (String cipher : ciphers) {
-            setSSLCipher(cipher, true);
+            try {
+                if (soc == null)
+                    setSSLCipher(cipher, true);
+                else
+                    setSSLCipher(soc, cipher, true);
+            } catch (Exception e) {
+                logger.debug(method + cipher + " failed to be set: " + e.toString());
```

Comment:
So are we wanting to treat this is a debug warning or would stronger action be called for?

Reply:
I don't think a stronger action is needed.  There are usually other ciphers that can be set and used.  However, if there isn't, then problem will occur as no connection can be made, and the admin will hear about it.